lang/rust/rust.py

```diff
@@ -0,0 +1,185 @@
+from talon import Module, Context, actions, ui, imgui, settings
+
+mod = Module()
+
+ctx = Context()
+ctx.matches = r'''
+mode: user.rust
+mode: user.auto_lang
+and code.language: rust
+'''
+
+ctx.tags = ['user.code_operators', 'user.code_generic']
+ctx.lists['user.code_libraries'] = {
+    'eye oh': 'std::io',
+    'file system': 'std::fs',
+    'envy': 'std::env',
+    'collections': 'std::collections',
+}
+ctx.lists['self.builtin_types'] = {
+    'you eight': 'u8',
+    'eye eight': 'i8',
+    'you sixteen': 'u16',
+    'eye sixteen': 'i16',
+    'you thirty two': 'u32',
+    'eye thirty two': 'i32',
+    'you sixty four': 'u64',
+    'eye sixty four': 'i64',
+    'you size': 'usize',
+    'eye size': 'isize',
+    'boolean': 'bool',
+}
+
+mod.list('builtin_types', desc='Rust builtin types')
+mod.list('code_libraries', desc='Rust std libraries')
```

Comment:
Don't need to declare code_libraries here since it's already defined in the `libraries_gui.py` file. Just the ctx version is fine.

I'd suggest using the existing code_type list as well rather than defining builtin_types. That way you can benefit from the stuff in functions.talon.